stdlib/unittest/mock.pyi

```diff
@@ -176,7 +175,10 @@ class _patch(Generic[_T]):
         kwargs: Mapping[str, Any],
     ) -> None: ...
     def copy(self) -> _patch[_T]: ...
-    def __call__(self, func: Callable[..., _R]) -> Callable[..., _R]: ...
+    @overload
+    def __call__(self, func: _TT) -> _TT: ...
+    @overload
+    def __call__(self, func: _F) -> _F: ...
```

Comment:
This is incorrect as the signature of the returned callable changes from the callable passed in. I think we should use the original definition for the second overload.

Reply:
Okay I was wondering about that. I think this comment implies that the current `decorate_callable` annotation is also incorrect so I've updated that, as well.